drivers/unix/os_unix.cpp

```diff
@@ -720,14 +728,40 @@ int OS_Unix::get_process_id() const {
 }
 
 bool OS_Unix::is_process_running(const ProcessID &p_pid) const {
+	const ProcessInfo *pi = process_map->get_ptr(p_id);
+	if (pi && !pi->is_running) {
+		return false;
+	}
+
 	int status = 0;
 	if (waitpid(p_pid, &status, WNOHANG) != 0) {
+		if (pi) {
+			pi->is_running = false;
+			pi->exit_code = status;
+		}
 		return false;
 	}
 
 	return true;
 }
 
+int OS_Unix::get_process_exit_code(const ProcessID &p_pid) const {
+	const ProcessInfo *pi = process_map->get_ptr(p_id);
+	if (pi && !pi->is_running) {
+		return pi->exit_code;
```

Comment:
```suggestion
		return WIFEXITED(pi->exit_code) ? WEXITSTATUS(pi->exit_code) : pi->exit_code;
```

`status` have additional bits set to indicate how it was terminated, since it's stored as is, mask should be applied (here or when storing it).